### PR TITLE
[Compute] BREAKING CHANGE: `az snapshot`: Migrate commands using Code Gen V2

### PR DESCRIPTION
**Pull request by yanzhudd**
**Related command**

`az snapshot create/update/grant-access`

**Description**

Migrate `snapshot` command group to Code Gen V2
Note: The subcommand groups have not yet been migrated

AAZ PR: https://github.com/Azure/aaz/pull/642

**Testing Guide**

**History Notes**

[Compute] BREAKING CHANGE: `az snapshot`: Refine output fields to align with backend service

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/vm/commands.py [hunk 1]
```diff
@@ -312,8 +312,8 @@ def load_command_table(self, _):
 
     with self.command_group('snapshot', compute_snapshot_sdk, operation_group='snapshots', min_api='2016-04-30-preview') as g:
         g.custom_command('create', 'create_snapshot', validator=process_snapshot_create_namespace, supports_no_wait=True)
-        g.custom_command('grant-access', 'grant_snapshot_access')
-        g.generic_update_command('update', custom_func_name='update_snapshot', setter_name='begin_create_or_update', setter_arg_name='snapshot', supports_no_wait=True)
+        from .operations.snapshot import SnapshotUpdate
+        self.command_table['snapshot update'] = SnapshotUpdate(loader=self)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/commands.py:316]: >         g.custom_command('grant-access', 'grant_snapshot_access')

Why do we need to remove `grant-access`?

File: src/azure-cli/azure/cli/command_modules/vm/operations/snapshot.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    def pre_instance_update(self, instance):
+        from azure.mgmt.core.tools import resource_id, is_valid_resource_id
+        from azure.cli.core.commands.client_factory import get_subscription_id
+
+        args = self.ctx.args
+        if has_value(args.disk_encryption_set):
+            if instance.properties.encryption.type != 'EncryptionAtRestWithCustomerKey' and \
+                    has_value(args.encryption_type) and \
+                    args.encryption_type != 'EncryptionAtRestWithCustomerKey':
+                raise CLIError('usage error: Please set --encryption-type to EncryptionAtRestWithCustomerKey')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/operations/snapshot.py:44]: Could we use specific error type instead od CLIError?

File: src/azure-cli/azure/cli/command_modules/vm/azure_stack/profile_2020_09_01_hybrid/operations/snapshot.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    def pre_instance_update(self, instance):
+        from azure.mgmt.core.tools import resource_id, is_valid_resource_id
+        from azure.cli.core.commands.client_factory import get_subscription_id
+
+        args = self.ctx.args
+        if has_value(args.disk_encryption_set):
+            if instance.properties.encryption.type != 'EncryptionAtRestWithCustomerKey' and \
+                    has_value(args.encryption_type) and \
+                    args.encryption_type != 'EncryptionAtRestWithCustomerKey':
+                raise CLIError('usage error: Please set --encryption-type to EncryptionAtRestWithCustomerKey')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/azure_stack/profile_2020_09_01_hybrid/operations/snapshot.py:41]: Same as above

File: src/azure-cli/azure/cli/command_modules/vm/operations/snapshot.py (hunk 2 above)
Comment (yanzhudd) [reply src/azure-cli/azure/cli/command_modules/vm/operations/snapshot.py]: It is not new logic for this PR and just moved in from custom.py file.
But you are right, I'll change it to `ArgumentUsageError`.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**